web/ui/src/Hosts/HostsController.js

```diff
@@ -129,6 +129,8 @@
         // update hosts
         update();
 
+        hostsFactory.activate();
```

Comment:
When do we deactivate the hostsFactory?  Does angular have a hook for leaving a page so we can deactivate a factory when the corresponding view is no longer active?